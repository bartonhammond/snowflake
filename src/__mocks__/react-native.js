/**
 * # __mockes__/react-native.js
 * 
 * This class stubs out the React-Native classes with React classes
 */
'use strict';
/**
 * ## Imports
 * 
 * ReactNative is actually React
 */ 
const React = require('react');
const ReactNative = React;

/**
 * ## These need additional mocking
 * 
 * ReactNative is actually React
 */ 
ReactNative.StyleSheet = {
    create: function create(styles) {
        return styles;
    }
};
class View extends React.Component {
    render() { return false; }
}
class PixelRatio extends React.Component {
    static get() { return 1; }
}
/**
 * ## Stubs
 * 
 * Simple replacements for testing
 */ 
ReactNative.View = View;
ReactNative.ScrollView = View;
ReactNative.Text = View;
ReactNative.TouchableOpacity = View;
ReactNative.TouchableHighlight = View;
ReactNative.TouchableWithoutFeedback = View;
ReactNative.ToolbarAndroid = View;
ReactNative.Image = View;
ReactNative.PixelRatio = PixelRatio;
ReactNative.NativeModules= {};

ReactNative.Platform = {};
module.exports = ReactNative;
