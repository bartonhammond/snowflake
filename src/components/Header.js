/**
 * # Header.js
 *
 * This component initially displays a image. But when clicked, things
 * get interesting.
 *
 * On the initial display after being clicked, the
 * textinput will display the current ```state``` of the application.
 *
 * The button will be enabled and if clicked, whatever state is now
 * contained in the textinput will be processed and the application
 * will be restored to that state.
 *
 * By pasting in a previous state, the application will reset to that
 * state
 *
 * When the mark image is clicked, it is just toggled to display or hide.
*/
'use strict'

/**
 * ## Imports
 *
 * React
*/
import React, {PropTypes} from 'react'
import
{
  ActivityIndicator,
  Image,
  StyleSheet,
  Text,
  TextInput,
  TouchableHighlight,
  View
} from 'react-native'

/**
 * Project component that will respond to onPress
 */
const FormButton = require('./FormButton')
/**
 * ## Styles
 */
var styles = StyleSheet.create({
  container: {
    flexDirection: 'column',
    flex: 1,
    marginTop: 10
  },
  header: {
    marginTop: 20,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: 'transparent'
  },
  mark: {
    height: 100,
    width: 100
  }

})

/**
 * ### Translations
 */
var I18n = require('react-native-i18n')
import Translations from '../lib/Translations'
I18n.translations = Translations

var Header = React.createClass({
  /**
   * ## Header.class
   * set the initial state of having the button be disabled.
   */
  getInitialState () {
    return {
      text: '',
      isDisabled: true
    }
  },
  /**
   * ### propTypes
   * * isFetching: display the spinner if true
   * * showState: should the JSON state, currentState, be displayed
   * * currentState: the JSON state
   * * onGetState: the action to call to get the current state
   * * onSetState: the action to call to set the state
   */
  propTypes: {
    isFetching: PropTypes.bool,
    showState: PropTypes.bool,
    currentState: PropTypes.object,
    onGetState: PropTypes.func,
    onSetState: PropTypes.func
  },
  /**
   * ### _onPressMark
   * Call the onGetState action passing the state prop
   */
  _onPressMark () {
    this.props.onGetState(!this.props.showState)
  },
  /**
   * ### _onChangeText
   * when the textinput value changes, set the state for that component
   */
  _onChangeText (text) {
    this.setState({
      text,
      isDisabled: false
    })
  },
  /**
   * ### _updateStateButtonPress
   * When the button for the state is pressed, call ```onSetState```
   */
  _updateStateButtonPress () {
    this.props.onSetState(this.state.text)
  },

  /**
   * ### render
   *
   * if showState, stringify the currentState and display it to the
   * browser for copying. Then display to the user.
   *
   * When the value of the input changes, call ```_onChangeText```
   *
   * When the 'Update State' button is pressed, we're off to the
   * races with Hot Loading...just call the
   * ```_updateStateButtonPress``` and away we go...
   *
   */
  render () {
    let displayText
    if (this.props.showState) {
      displayText = JSON.stringify(this.props.currentState)
    }

    return (
      <View>
        <View style={styles.header}>

          <TouchableHighlight onPress={this._onPressMark}>

            <Image style={styles.mark}
              source={require('../images/Snowflake.png')}
            />
          </TouchableHighlight>
          {this.props.isFetching
           ? <ActivityIndicator animating size='large' />
           : null
          }

        </View>
        {this.props.showState
         ? <View style={styles.container}>
           <Text>{I18n.t('Header.current_state')} ({I18n.t('Header.see_console')})</Text>
           <TextInput style={{height: 100, borderColor: 'gray', borderWidth: 1}}
             value={displayText}
             editable
             multiline
             onChangeText={(text) => this._onChangeText(text)}
             numberOfLines={20} />
           <View style={{
             marginTop: 10
           }}>
             <FormButton isDisabled={this.state.isDisabled}
               onPress={this._updateStateButtonPress}
               buttonText={I18n.t('Header.update_state')} />
           </View>
         </View>
         : null}
      </View>
    )
  }
})

module.exports = Header
