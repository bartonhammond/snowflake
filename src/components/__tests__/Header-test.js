/**
 * # Header-test.js
 *
 * This class tests that the Header component displays correctly
 *
 * *Note:* if you want to understand the structures here, add a
 * ```console.log``` and then ```npm test```.
 *
 */
'use strict'

jest.mock('ActivityIndicator', () => 'ActivityIndicator')

/**
* ## Imports
*/
import 'react-native'
import React from 'react'

import Header from '../Header'

import ReactTestUtils from 'react-addons-test-utils'
const renderer = ReactTestUtils.createRenderer()

/**
 * ## Test
 */
describe('Header', () => {
  /**
   * ### it should be display empty text when not fetching
   * render the header when not fetching
   */
  it('should be display empty text when not fetching', () => {
    const props = {
      isFetching: false
    }
    renderer.render(<Header {...props} />)
    const tree = renderer.getRenderOutput()
    expect(tree).toMatchSnapshot()
  })
  /**
   * ### it should be display spinner when fetching
   * When fetching, the GiftedSpinner should display
   */
  it('should be display spinner when fetching', () => {
    const props = {
      isFetching: true
    }
    renderer.render(<Header {...props} />)
    const tree = renderer.getRenderOutput()
    expect(tree).toMatchSnapshot()
  })
})// describe Header
