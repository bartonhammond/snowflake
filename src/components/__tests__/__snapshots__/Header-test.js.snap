exports[`Header should be display empty text when not fetching 1`] = `
<View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "justifyContent": "center",
        "marginTop": 20,
      }
    }>
    <TouchableHighlight
      activeOpacity={0.85}
      onPress={[Function]}
      underlayColor="black">
      <Image
        source={1}
        style={
          Object {
            "height": 100,
            "width": 100,
          }
        } />
    </TouchableHighlight>
  </View>
</View>
`;

exports[`Header should be display spinner when fetching 1`] = `
<View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "justifyContent": "center",
        "marginTop": 20,
      }
    }>
    <TouchableHighlight
      activeOpacity={0.85}
      onPress={[Function]}
      underlayColor="black">
      <Image
        source={1}
        style={
          Object {
            "height": 100,
            "width": 100,
          }
        } />
    </TouchableHighlight>
    <ActivityIndicator
      animating={true}
      size="large" />
  </View>
</View>
`;
